cmd/gcp-controller-manager/kubelet_readonly_csr_approver_test.go

```diff
@@ -112,22 +106,58 @@ func TestKubeletReadonlyApprover(t *testing.T) {
+			expectCondition: capi.CertificateSigningRequestCondition{
+				Type:    capi.CertificateDenied,
+				Reason:  "AutoDenied",
+				Message: "csr usage any is not allowed, allowed usages: [\"key encipherment\",\"digital signature\",\"client auth\"]",
+				Status:  v1.ConditionTrue,
+			},
+		},
+		{
+			name:                   "fail by pod annotation validation error",
+			sarAllowed:             true,
+			podAnnotationAllowed:   true,
+			usageAllowed:           true,
+			autopilotEnabled:       true,
+			podAnnotationwantError: true,
+			expectError:            fmt.Errorf("validating CSR \"fail by pod annotation validation error\" failed: validator \"pod annotation validator\" has error get pod test failed: failed to get pod"),
+			expectCondition: capi.CertificateSigningRequestCondition{
+				Type:    capi.CertificateDenied,
+				Reason:  "AutoDenied",
+				Message: "csr usage any is not allowed, allowed usages: [\"key encipherment\",\"digital signature\",\"client auth\"]",
+				Status:  v1.ConditionTrue,
+			},
+		},
 	}
 
 	for _, testCase := range testCases {
 		t.Run(fmt.Sprint(testCase.name), func(t *testing.T) {
-			request := generateTestRequest(t, testCase.sarAllowed, testCase.podAnnotationAllowed, testCase.usageAllowed, testCase.autopilotEnabled)
+			request := generateTestRequest(t, testCase.sarAllowed, testCase.podAnnotationAllowed, testCase.usageAllowed, testCase.autopilotEnabled,
+				testCase.sarwantError, testCase.podAnnotationwantError)
+			request.csr.Name = testCase.name
 			request.csr.Spec.SignerName = kubeletReadonlyCSRSignerName
 			approver := newKubeletReadonlyCSRApprover(request.controllerContext)
 			*autopilotEnabled = testCase.autopilotEnabled
 			err := approver.handle(request.context, request.csr)
-			if err != nil {
-				t.Fatalf("error when handle approver, error: %v", err)
-			}
-			if diff := cmp.Diff(request.csr.Status.Conditions[0], testCase.expectCondition, cmp.Comparer(func(x, y capi.CertificateSigningRequestCondition) bool {
-				return x.Type == y.Type && x.Reason == y.Reason && x.Message == y.Message && x.Status == y.Status
-			})); diff != "" {
-				t.Fatalf("condition don't match, diff -want +got\n%s", diff)
+			switch {
+			case testCase.expectError == nil && err == nil:
+				if diff := cmp.Diff(request.csr.Status.Conditions[0], testCase.expectCondition, cmp.Comparer(func(x, y capi.CertificateSigningRequestCondition) bool {
+					return x.Type == y.Type && x.Reason == y.Reason && x.Message == y.Message && x.Status == y.Status
+				})); diff != "" {
+					t.Fatalf("condition don't match, diff -want +got\n%s", diff)
+				}
+			case testCase.expectError != nil && err == nil ||
+				err != nil && testCase.expectError == nil ||
+				err.Error() != testCase.expectError.Error():
+				t.Fatalf("error not match, expect: %v, got: %v", testCase.expectError, err)
```

Comment:
In go got is preferred before wat, So this error message should be

t.Fatal("got error: %v; want error: %v", err, testCase.expectedError)